examples/eestream/serve-collected/main.go

```diff
@@ -46,13 +52,24 @@ func Main() error {
 	if err != nil {
 		return err
 	}
+	// initialize http rangers in parallel to save from network latency
 	rrs := map[int]ranger.Ranger{}
-	for i := 0; i < 40; i++ {
-		url := fmt.Sprintf("http://localhost:%d", 10000+i)
-		rrs[i], err = ranger.HTTPRanger(url)
-		if err != nil {
+	result := make(chan indexRangerError, *rsn)
+	for i := 0; i < *rsn; i++ {
+		go func(i int) {
+			url := fmt.Sprintf("http://localhost:%d", 10000+i)
+			rr, err := ranger.HTTPRanger(url)
```

Comment:
oh right cause of the HEAD requests! good catch!

one thing i've been thinking is that as we get this closer to production, i expect we'll store the size of each shard in the pointer. so we might already have the size in the codepath here and can avoid a HEAD request.

maybe in the future it will make sense to have something like an ranger.PresizedHTTPRanger(url, size) or something?

but yeah this is great

Reply:
yes, we can add such constructor when it make sense.

Reply:
i wonder how much knowing the size upfront would improve performance even with this test